src/Candidate.js

```diff
@@ -1,31 +1,44 @@
 import React from 'react';
 import styled from 'styled-components';
 import axios from 'axios';
+import { useHistory } from 'react-router-dom';
+
 const url =
   'http://ec2-13-209-5-166.ap-northeast-2.compute.amazonaws.com:8000/api/vote?';
 
-function CandidateVotes({ candidate, flipVoteFlag, rank }) {
+function CandidateVotes({ candidate, flipVoteFlag, rank, loginCookie }) {
+  let history = useHistory();
   function handleVoteButtonClick() {
     axios
       .get(url, {
         params: {
           id: candidate.id,
         },
+        headers: {
+          authorization: `${loginCookie.loginCookie}`,
+        },
       })
       .then(function (response) {
         alert(response.data);
         flipVoteFlag();
       })
       .catch(function (error) {
-        console.log(error);
+        if (error.response.status === 401) {
+          alert('로그인 후 투표 가능합니다.');
+          history.push('/signin');
+        } else if (error.response.status === 404)
+          alert('해당 후보는 존재하지 않습니다.');
+        else alert(error.response.data);
       });
   }
   return (
     <CandidateContainer>
```

Comment:
```suggestion
    <Wrapper>
```
위에서 말한것처럼 Wrapper 사용해보면 좋을듯!